crates/router/src/core/payment_link.rs

```diff
@@ -586,6 +602,11 @@ pub fn get_payment_link_config_based_on_priority(
         display_sdk_only,
         enabled_saved_payment_method,
         allowed_domains,
+        merchant_details: serde_json::to_string(&merchant_details)
+            .change_context(errors::ApiErrorResponse::InvalidDataValue {
+                field_name: "merchant_details",
+            })
+            .ok(),
     };
```

Comment:
What is scope of this field is this transactional data, if yes then it should only be in paymentscreate payment link config. And if that is the case then we donot need this field to go through priority macro